plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile

```diff
@@ -827,6 +827,65 @@ public static void main(String[] args) {
     statemachine.runCycle();
 }
 
+p. 
+
+h4(#codegen_java_trace-observed_state_machine). Trace-observed state machine
+
+p. By using the tracing feature the execution of the state machine can be observed. In detail, entered and exited states can be traced. Therefore, additional operation callbacks _onStateEntered_ and _onStateExited_ are generated in the _ITracingListener_ interface:
+
+bc.. 
+/**
+ * Tracing interface for state machines.
+ * Must be implemented if tracing feature is used.
+ * Multiple listeners can be added to and removed from the state machine.
+ */
+public interface ITracingListener<T> {
+	
+	/**
+	 * OnStateEntered will be called if any state has been entered.
+	 */
+	void onStateEntered(T state);
+	
+	/**
+	* OnStateExited will be called if any state has been exited.
+	*/
+	void onStateExited(T state);
+}
+
+p. With additional generated API in the generated state machine, listeners can be added to and removed from the state machine. Therefore, a implementation of the _ITracingListener_ interface is required. A minimum implementation could be done like this:
+
+bc.. 
+public class TracingImpl<T> implements ITracingListener<T>{
+	@Override
+	public void onStateEntered(T state) {}
+	@Override
+	public void onStateExited(T state) {}
+}
+
+p. If the tracing feature is enabled, at least one listener must be added:
```

Comment:
Why? Is that checked somewhere? There's nothing wrong with iterating over an empty list.